plugins/netskope/plugin.spec.yaml

```
@@ -0,0 +1,413 @@
+plugin_spec_version: v2
+extension: plugin
+products: [insightconnect]
+name: netskope
+title: Netskope
+description: This plugin allows users to create, read, update, and delete URL list
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+supported_versions: ["2022-02-11"]
+status: []
+resources:
+  source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/plugins/netskope
+  license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://www.netskope.com/
+tags:
+  - netskope
+  - cloud
+  - security
+hub_tags:
+  use_cases: [cloud_security]
+  keywords: [endpoint]
+  features: []
+enable_cache: false
+connection:
+  tenant:
+    title: Tenant
+    description: Owner's name that is included in URL
+    type: string
+    required: true
+    example: exampletenant
+  api_key_v1:
+    title: API Key v1
+    description: Netskope Authorization Token
+    type: credential_secret_key
+    required: true
+    example: j5740ps1cbukyk3t8kib3wa36aq2v3da
+  api_key_v2:
+    title: API Key v2
+    description: Netskope Authorization Token
+    type: credential_secret_key
+    required: true
+    example: vfcl7japffxtlna0zugmlxrsxkhe3lnc
+actions:
+  update_file_hash_list:
+    title: Update a File Hash List
+    description: Updates the File Hash List with the values provided
+    input:
+      name:
+        title: Name
+        description: Name of an existing File Hash List shown in the Netskope UI on the File Hash List page
+        type: string
+        required: true
+        example: ExampleExistingFileHashList
+      list:
+        title: List
+        description: Comma-separated list of file hashes (md5 or sha256), with the maximum size of the list being 8MB
+        type: "[]string"
+        required: true
+        example: '["e28eb9739b6e84d0f796e3acc0f5b71e", "e28eb9739b6e84d0f697e3acc0f5b71a", "e28eb9839b6e74d0f696e3acc0f6b710"]'
+    output:
+      status:
+        title: Status
+        description: API response status
+        type: string
+        required: false
+      message:
+        title: Message
+        description: API response message
+        type: string
+        required: false
+  get_all_url_list:
+    title: Get All URL List
+    description: Get all applied or pending URL lists
+    input:
+      status:
+        title: Status
+        description: Status of URL lists to be received
+        type: string
+        enum:
+          - ""
+          - applied
+          - pending
+        required: false
+        example: applied
+    output:
+      urllists:
+        title: URL Lists
+        description: Contains list of urllist objects
+        type: "[]object"
```

I am pretty sure, but not 100% sure, that it is preferred not to enquote types that begin with an array indicator.

```suggestion
        type: []object
```

---

Well if I remove the enquote there it makes that some errors occur while regenerating the plugin - with them included there are no errors.